### ISSUE TITLE
Performance discrepancy when reproducing results on Re10K

### ISSUE THREAD

**Issue by Yuhuoo** [ziplab/VolSplat#19]
First of all, thank you for the impressive work and for open-sourcing the code!

I attempted to reproduce the training results using the default configuration without any code modifications. However, I noticed a significant performance gap between my trained model and the provided checkpoints.

Training Command:
```
CUDA_VISIBLE_DEVICES=0 python -m src.main +experiment=re10k \
data_loader.train.batch_size=1 \
'dataset.roots'='["datasets/re10k"]' \
dataset.test_chunk_interval=50 \
dataset.num_context_views=6 \
trainer.max_steps=150000 \
model.encoder.num_scales=2 \
model.encoder.upsample_factor=2 \
model.encoder.lowest_feature_resolution=4 \
model.encoder.monodepth_vit_type=vitb \
output_dir=outputs-vitb/re10k-256x256-vitb \
wandb.mode=disabled \
checkpointing.pretrained_monodepth=pretrained/depth_anything_v2_vitb.pth \
checkpointing.pretrained_mvdepth=pretrained/gmflow-scale1-things-e9887eda.pth \
2>&1 | tee re10k-256x256-volsplat-vitb.txt
```

After the training was completed, I ran the evaluation using the following command:
```
python -m src.main +experiment=re10k \
data_loader.train.batch_size=1 \
'dataset.roots'='["datasets/re10k"]' \
dataset.test_chunk_interval=10 \
dataset/view_sampler=evaluation \
dataset.view_sampler.num_context_views=6 \
dataset.view_sampler.index_path=assets/re10k_evaluation/evaluation_index_re10k.json \
trainer.max_steps=150000 \
model.encoder.num_scales=2 \
model.encoder.upsample_factor=2 \
model.encoder.lowest_feature_resolution=4 \
model.encoder.monodepth_vit_type=vitb \
mode=test \
test.save_video=false \
test.save_depth_concat_img=false \
test.save_image=false \
test.save_gt_image=false \
test.save_input_images=false \
test.save_video=false \
test.save_gaussian=false \
checkpointing.pretrained_model=outputs-vitb/re10k-256x256-vitb/checkpoints/epoch_2-step_150000.ckpt \
output_dir=outputs-vitb-eval/ori-volsplat-re10k-256x256-test
```

The results are as follows:

`{"psnr": 29.472512093182175, "ssim": 0.9207621057884313, "lpips": 0.09319586187012609, "encoder": [704, 0.15365556898442181], "decoder": [5632, 0.0022843294268981977]}`

For comparison, testing with your provided checkpoints yielded the following results:

`{"psnr": 31.213006693486268, "ssim": 0.9399100188980318, "lpips": 0.07484684167337854, "encoder": [704, 0.16290524195541034], "decoder": [5632, 0.002489270473068411]}`

Is it possible that some hyperparameters or environment settings were not configured correctly?
Thank you in advance for your help! Looking forward to your reply.

**Comment by Alresd** [comment]
> First of all, thank you for the impressive work and for open-sourcing the code!
> 
> I attempted to reproduce the training results using the default configuration without any code modifications. However, I noticed a significant performance gap between my trained model and the provided checkpoints.
> 
> Training Command:
> 
> ```
> CUDA_VISIBLE_DEVICES=0 python -m src.main +experiment=re10k \
> data_loader.train.batch_size=1 \
> 'dataset.roots'='["datasets/re10k"]' \
> dataset.test_chunk_interval=50 \
> dataset.num_context_views=6 \
> trainer.max_steps=150000 \
> model.encoder.num_scales=2 \
> model.encoder.upsample_factor=2 \
> model.encoder.lowest_feature_resolution=4 \
> model.encoder.monodepth_vit_type=vitb \
> output_dir=outputs-vitb/re10k-256x256-vitb \
> wandb.mode=disabled \
> checkpointing.pretrained_monodepth=pretrained/depth_anything_v2_vitb.pth \
> checkpointing.pretrained_mvdepth=pretrained/gmflow-scale1-things-e9887eda.pth \
> 2>&1 | tee re10k-256x256-volsplat-vitb.txt
> ```
> 
> After the training was completed, I ran the evaluation using the following command:
> 
> ```
> python -m src.main +experiment=re10k \
> data_loader.train.batch_size=1 \
> 'dataset.roots'='["datasets/re10k"]' \
> dataset.test_chunk_interval=10 \
> dataset/view_sampler=evaluation \
> dataset.view_sampler.num_context_views=6 \
> dataset.view_sampler.index_path=assets/re10k_evaluation/evaluation_index_re10k.json \
> trainer.max_steps=150000 \
> model.encoder.num_scales=2 \
> model.encoder.upsample_factor=2 \
> model.encoder.lowest_feature_resolution=4 \
> model.encoder.monodepth_vit_type=vitb \
> mode=test \
> test.save_video=false \
> test.save_depth_concat_img=false \
> test.save_image=false \
> test.save_gt_image=false \
> test.save_input_images=false \
> test.save_video=false \
> test.save_gaussian=false \
> checkpointing.pretrained_model=outputs-vitb/re10k-256x256-vitb/checkpoints/epoch_2-step_150000.ckpt \
> output_dir=outputs-vitb-eval/ori-volsplat-re10k-256x256-test
> ```
> 
> The results are as follows:
> 
> `{"psnr": 29.472512093182175, "ssim": 0.9207621057884313, "lpips": 0.09319586187012609, "encoder": [704, 0.15365556898442181], "decoder": [5632, 0.0022843294268981977]}`
> 
> For comparison, testing with your provided checkpoints yielded the following results:
> 
> `{"psnr": 31.213006693486268, "ssim": 0.9399100188980318, "lpips": 0.07484684167337854, "encoder": [704, 0.16290524195541034], "decoder": [5632, 0.002489270473068411]}`
> 
> Is it possible that some hyperparameters or environment settings were not configured correctly? Thank you in advance for your help! Looking forward to your reply.

Hi ,

Thank you for your interest in our work and for providing such detailed information about your reproduction steps.

After reviewing your training command, I noticed that you are training on a single GPU with data_loader.train.batch_size=1. As mentioned in the Implementation Details (Section 4.1) of our paper:

"...we train the model for 150,000 iterations using 4× NVIDIA H20 GPUs with a total batch size of 4."

The gap in performance (approx. 1.7 dB PSNR) is likely due to this difference in the effective batch size. In our experiments, a total batch size of 4 provides more stable gradient estimates. 

I hope this helps! Feel free to let me know if you have further questions.

**Comment by lhmd** [comment]
Closing as there has been no further activity. Feel free to reopen if needed.
